### PR TITLE
Adds unit test for spawned food edibility

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

I made two PRs recently (#57205, #57179) fixing food that when spawned by an admin or the map was inedible due to a lack of reagents. This adds a unit test to make sure spawned food gets reagents. 

It also checks for the edible component, as a bug with butchering (#56126) results in butchered meat becoming inedible due to a lack of the component. While butchering works differently to simply spawning the meat, it is a foreseeable potential issue.

**Also does the following:**
- Adds stack trace for biting food failing due to a lack of reagents
~~- Adds checks for seed var having been set in the initialize() of grown food to avoid runtimes (one of which was preventing this unit test from succeeding)~~ Checking this here just results in later runtimes and unpredictable behaviour, if it runtimes a seed should be added
- Adds seeds for bombananas (it was causing a runtime ~~which the above fixes, but this is probably better than them not having seeds~~ because produce uses its seed for setting reagents and the like) -- ~~test this for unforeseen issues~~
- Fixes issue with gondola meat code duplicated for penguin meat slabs resulting in a runtime (which was preventing this unit test from succeeding) and being unable to slice penguin meat

## Why It's Good For The Game

Should reduce instances of food that can't be eaten.

## Changelog
:cl: cacogen
code: Should be less likely to get inedible food now
fix: Slicing penguin meat slabs into raw cutlets should work again
/:cl:

### CODE REVIEW

File: code/modules/unit_tests/food_reagents_check.dm [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+/datum/unit_test/food_edibility_check
+
+/datum/unit_test/food_edibility_check/Run()
+	var/list/not_food = list(
+	/obj/item/food/grown,
+	/obj/item/food/grown/mushroom,
+	/obj/item/food/deepfryholder,
+	/obj/item/food/clothing,
+	/obj/item/food/meat/slab/human/mutant,
+	/obj/item/food/grown/shell)
```
Comment (Mothblocks) [inline code/modules/unit_tests/food_reagents_check.dm:11]: Indent.

File: code/modules/unit_tests/food_reagents_check.dm [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+	var/list/food_paths = subtypesof(/obj/item/food) - not_food
+
+	for(var/obj/item/food/food_path as anything in food_paths)
+		var/obj/item/food/spawned_food = new food_path
+
+		if(!spawned_food.reagents)
+			Fail("[food_path] does not have any reagents, making it inedible!")
+
+		if(!spawned_food.GetComponent(/datum/component/edible))
```
Comment (Mothblocks) [inline code/modules/unit_tests/food_reagents_check.dm:21]:
```suggestion
		if(!IS_EDIBLE(spawned_food))
```

File: code/modules/unit_tests/food_reagents_check.dm [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	/obj/item/food/grown,
+	/obj/item/food/grown/mushroom,
+	/obj/item/food/deepfryholder,
+	/obj/item/food/clothing,
+	/obj/item/food/meat/slab/human/mutant,
+	/obj/item/food/grown/shell)
+
+	var/list/food_paths = subtypesof(/obj/item/food) - not_food
+
+	for(var/obj/item/food/food_path as anything in food_paths)
```
Comment (Mothblocks) [inline code/modules/unit_tests/food_reagents_check.dm:15]:
```suggestion
	for(var/food_path in food_paths)
```

You're not using `initial` or anything that would require this be typed.

File: code/modules/unit_tests/food_reagents_check.dm [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+		var/obj/item/food/spawned_food = new food_path
```
Comment (Mothblocks) [inline code/modules/unit_tests/food_reagents_check.dm:16]:
```suggestion
		var/obj/item/food/spawned_food = allocate(food_path)
```

`allocate` is something in tests that will automatically delete at the end. I know you're already using `qdel`, but `allocate` provides some good safety (what if GetComponent panics, for some reason).

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Minor test quips. Thanks!

File: code/modules/unit_tests/food_reagents_check.dm (hunk 1 above)
Comment (cacogen) [reply code/modules/unit_tests/food_reagents_check.dm:11]: Indent the list items by one tab?

**Comment by cacogen** [comment]
Note to self

- [ ] Indent not_food list
- [ ] Rename file to match unit test

File: code/modules/unit_tests/food_reagents_check.dm (hunk 1 above)
Comment (Mothblocks) [reply code/modules/unit_tests/food_reagents_check.dm:11]: Yep yep.

File: code/modules/hydroponics/grown.dm [hunk 5]
```diff
@@ -46,13 +46,13 @@
...
-	for(var/datum/plant_gene/trait/trait in seed.genes)
+	for(var/datum/plant_gene/trait/trait in seed?.genes)
 		trait.on_new_plant(src, loc)
 
 	. = ..() //Only call it here because we want all the genes and shit to be applied before we add edibility. God this code is a mess.
 
-	seed.prepare_result(src)
-	transform *= TRANSFORM_USING_VARIABLE(seed.potency, 100) + 0.5 //Makes the resulting produce's sprite larger or smaller based on potency!
+	seed?.prepare_result(src)
+	transform *= TRANSFORM_USING_VARIABLE(seed?.potency, 100) + 0.5 //Makes the resulting produce's sprite larger or smaller based on potency!
```
Comment (MrMelbert) [inline code/modules/hydroponics/grown.dm:55]: Do you know what happens to the transform if seed happens to be null in this line?

Also, which plant was failing the test for having a `null` seed? If there's a grown item with a null seed that might be a bigger problem

Comment (cacogen) [reply code/modules/hydroponics/grown.dm]: It was just the bombanana with a null seed. I tested it, and while it just spawned a tiny banana, it created a runtime with eating it. I imagine other runtimes with other growns potentially. So I just decided not to check it at all. If it runtimes, people should fix it by adding a seed.

**Comment by cacogen** [comment]
>PASS: /datum/unit_test/food_edibility_check 0.6s

```
[23:17:44] Runtime in beam.dm,87: Cannot read null.x
  proc name: Draw (/datum/beam/proc/Draw)
  src: /datum/beam (/datum/beam)
  call stack:
  /datum/beam (/datum/beam): Draw()
  /datum/beam (/datum/beam): Start()
  world: ImmediateInvokeAsync(/datum/beam (/datum/beam), /datum/beam/proc/Start (/datum/beam/proc/Start))
  space (122,17,12) (/turf/open/space/basic): Beam(null, "curse0", 'icons/effects/beam.dmi', 32, 1e+31, /obj/effect/ebeam/curse_arm (/obj/effect/ebeam/curse_arm))
  Hel\'s grasp (/obj/projectile/curse_hand/hel): Destroy(0)
  qdel(Hel\'s grasp (/obj/projectile/curse_hand/hel), 0)
  Hel\'s grasp (/obj/projectile/curse_hand/hel): pixel move(1, 0)
  Hel\'s grasp (/obj/projectile/curse_hand/hel): process(0.1)
  Projectiles (/datum/controller/subsystem/processing/projectiles): fire(0)
  Projectiles (/datum/controller/subsystem/processing/projectiles): ignite(0)
  Master (/datum/controller/master): RunQueue()
  Master (/datum/controller/master): Loop()
  Master (/datum/controller/master): StartProcessing(0)
```
```
[23:17:47] Runtime in _component.dm,219: Cannot read null.comp_lookup
  proc name: UnregisterSignal (/datum/proc/UnregisterSignal)
  src: /datum/beam (/datum/beam)
  call stack:
  /datum/beam (/datum/beam): UnregisterSignal(null, "movable_moved")
  /datum/beam (/datum/beam): Destroy(0)
  qdel(/datum/beam (/datum/beam), 0)
  /datum/callback (/datum/callback): InvokeAsync()
  Timer (/datum/controller/subsystem/timer): fire(0)
  Timer (/datum/controller/subsystem/timer): ignite(0)
  Master (/datum/controller/master): RunQueue()
  Master (/datum/controller/master): Loop()
  Master (/datum/controller/master): StartProcessing(0)
```

**Comment by Mothblocks** [comment]
Sporadic failure, not your fault.

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is now above zero (1). Feel free to make Feature/Tweak/Balance PRs.
